web_src/js/features/clipboard.js

```diff
@@ -1,21 +1,18 @@
 // For all DOM elements with [data-clipboard-target] or [data-clipboard-text], this copy-to-clipboard will work for them
 
-// TODO: replace these with toast-style notifications
 function onSuccess(btn) {
-  if (!btn.dataset.content) return;
   $(btn).popup('destroy');
   const oldContent = btn.dataset.content;
-  btn.dataset.content = btn.dataset.success;
+  btn.dataset.content = btn.dataset.success || '';
```

Comment:
Maybe we can use `btn.dataset.content = btn.dataset.clipboardCopySuccess || i18n.common_clipboard_copy_success;` here, then we can remove most redundant HTML attributes like `data-success` / `data-error`.

Reply:
Should be possible to generalize this, yes.